File: 2799. Count Complete Subarrays in an Array/countCompleteSubarrays.cpp
#include <unordered_set>
#include <unordered_map>
#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    int countCompleteSubarrays(vector<int>& nums) {
        int ans=0;
        int l=0;
        int sz = unordered_set<int>(nums.begin(), nums.end()).size();
        unordered_map<int,int>mp;

        for(int r=0; r<nums.size(); r++){
            mp[nums[r]]+=1;

            while(mp.size() == sz){
                ans += (nums.size() - r);
                mp[nums[l]] -= 1;
                if(mp[nums[l]] == 0)mp.erase(nums[l]);
                l++;
            }
        }

        return ans;
    }
};

File: 2799. Count Complete Subarrays in an Array/countCompleteSubarrays.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countCompleteSubarrays = function (nums) {
    let ans = 0;

    let l = 0;
    let unique = new Set([...nums]).size
    let mp = new Map();

    for (let r = 0; r < nums.length; r++) {
        if (!mp.has(nums[r])) {
            mp.set(nums[r], 0)
        }
        mp.set(nums[r], mp.get(nums[r]) + 1)

        while (mp.size === unique) {
            ans += (nums.length - r)
            mp.set(nums[l], mp.get(nums[l]) - 1)
            if (mp.get(nums[l]) === 0) {
                mp.delete(nums[l])
            }
            l++
        }
    }

    return ans
};